### PR TITLE
Added the NAVY Seal

### PR DESCRIPTION
**Pull request by Mailaender**
See http://cnc.wikia.com/wiki/Navy_SEAL_(Red_Alert_2) for reference.

### CODE REVIEW

File: rules/allied-infantry.yaml [hunk 1]
```diff
@@ -212,6 +212,54 @@ spy:
...
+		Bounds: 20, 30, 0, -11
+	Health:
+		HP: 125
+	Mobile:
+		Speed: 71
+	RevealsShroud:
+		Range: 8c0
+	C4Demolition:
+		C4Delay: 45
+		Voice: Attack
```
Comment (GraionDilach) [inline rules/allied-infantry.yaml]: This is wrong. https://github.com/Phrohdoh/oramod-ra2/blob/master/extra-files/sound.ini#L1179 are the sounds used for this.

Comment (Mailaender) [inline rules/allied-infantry.yaml]: Added the SpecialAttack voice.

File: rules/allied-infantry.yaml [hunk 2]
```diff
@@ -212,6 +212,54 @@ spy:
...
+	Buildable:
+		Queue: Infantry
+		BuildAtProductionType: Infantry
+		BuildPaletteOrder: 110
+		Prerequisites: ~disabled
+	Valued:
+		Cost: 1000
+	Tooltip:
+		Name: Navy SEAL
+		Description: Elite commando infantry, armed with \na sub machine gun and C4.\n  Strong vs Infantry, Buildings\n  Weak vs Vehicles, Aircraft\nSpecial Ability: Destroy Building with C4
```
Comment (abcdefg30) [inline rules/allied-infantry.yaml]: You can drop the space before the `\n`.

Comment (Mailaender) [inline rules/allied-infantry.yaml]: Done.

File: rules/allied-infantry.yaml [hunk 3]
```diff
@@ -212,6 +212,54 @@ spy:
...
+		UpgradeMaxAcceptedLevel: 1
+	Armament@elite:
+		Weapon: MP5E
+		UpgradeTypes: eliteweapon
+		UpgradeMinEnabledLevel: 1
+	AttackFrontal:
+		Voice: Attack
+	RenderSprites:
+		Image: seal
+	WithInfantryBody:
```
Comment (abcdefg30) [inline rules/allied-infantry.yaml]: Sorry to bother you again. But this is already present on `^Infantry` (and thus shouldn't be explicitly needed here).

File: sequences/allied-infantry.yaml [hunk 4]
```diff
@@ -288,6 +288,60 @@ spy:
...
+	die1:
+		Start: 134
+		Length: 14
+	die2:
+		Start: 149
+		Length: 14
+	die4: nukedie
+		Length: *
+		Tick: 50
+		# Problem: Palette = effect
```
Comment (abcdefg30) [inline sequences/allied-infantry.yaml]: Shouldn't be a problem anymore: https://github.com/OpenRA/ra2/blob/master/rules/defaults.yaml#L278

**Comment by abcdefg30** [comment]
Are you going to add the passenger upgrade for the IFV in this pr, or do you want to leave that for a follow-up?

**Comment by Mailaender** [comment]
Rebased due to https://github.com/OpenRA/ra2/pull/78 and updated.

**Comment by abcdefg30** [comment]
The IFV turret fire report is tanyas which is a bit unpolished imho, but not a blocker.

Good work. :+1:

**Comment by jtyle6** [comment]
Only can be trained with Build Everything toggled on.

**Comment by MustaphaTR** [comment]
Because SEALs are not buildable in RA2 they are mission specific unit. They are buildable only in YR.

**Comment by GraionDilach** [comment]
The SEAL in RA2 is a male Tanya. Since Tanya doesn't have a buildlimit, there's no point in training them even.

In YR Tanya got a buildlimit and C4 against vehicles which "justified" the Navy SEAL's availability.

**Comment by jtyle6** [comment]
I have played yr.  And played the Allied campaign on ra2. Toggle for yr units.

**Pull request merged**